### PR TITLE
Histogram

### PR DESCRIPTION
**Pull request by emmanuelmathot**
`histogram` field for the histogram distribution information of the pixels values in the band

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -145,3 +146,31 @@ In the following value definition example, 185 meters must be substracted from t
...
+| Field Name | Type      | Description                                                                 |
+| ---------- | --------- | --------------------------------------------------------------------------- |
+| count      | number    | number of buckets of the distribution.                                      |
+| min        | number    | minimum value of the distribution. Also the mean value of the first bucket. |
+| max        | number    | minimum value of the distribution. Also the mean value of the last bucket.  |
+| buckets    | \[number] | Array of integer indicating the number of pixels included in the bucket.    |
+
+The information in histogram object may be useful to prepare a user interface in the perspective of the manipulation of the pixels value
+for raster visualization.
+For instance to enhances an image by changing properties such as brightness, contrast, and gamma through multiple stretch types
```
Comment (kylebarron) [inline README.md:163]:
```suggestion
For instance to enhance an image by changing properties such as brightness, contrast, and gamma through multiple stretch types
```

**Review by kylebarron — COMMENTED** [review]
No real objections to this.

- Are the values in `buckets` always raw counts rather than a percentage of the total number of pixels?
- Presumably the buckets are always equal-size? I could imagine instances where a lot of the values are in a few buckets, and you'd like to have more detail about that specific range, but it would probably be harder to convey that in a json format like this.

**Comment by emmanuelmathot** [comment]
> No real objections to this.
> 
>     * Are the values in `buckets` always raw counts rather than a percentage of the total number of pixels?
Yes, that is how it comes out of gdalinfo. That is also the case when you export histogram information in QGIS
> 
>     * Presumably the buckets are always equal-size?
Yes, the bucket width are all equals and mainly depends on the number of buckets.
it can be computed with `Bucket width = (Max-Min) ÷ Number of Buckets`
>
>  I could imagine instances where a lot of the values are in a few buckets, and you'd like to have more detail about that specific range, but it would probably be harder to convey that in a json format like this.

In that case, the sampling must be finer and the number of buckets increased. That histogram is for a sketchy analysis of the distribution. When you want more info about specific values, the data must be accessed

File: README.md [hunk 2]
```diff
@@ -145,3 +146,31 @@ In the following value definition example, 185 meters must be substracted from t
...
+| ---------- | --------- | --------------------------------------------------------------------------- |
+| count      | number    | number of buckets of the distribution.                                      |
+| min        | number    | minimum value of the distribution. Also the mean value of the first bucket. |
+| max        | number    | minimum value of the distribution. Also the mean value of the last bucket.  |
+| buckets    | \[number] | Array of integer indicating the number of pixels included in the bucket.    |
+
+The information in histogram object may be useful to prepare a user interface in the perspective of the manipulation of the pixels value
+for raster visualization.
+For instance to enhance an image by changing properties such as brightness, contrast, and gamma through multiple stretch types
+such as statistical functions.
```
Comment (philvarner) [inline README.md:164]: You could also add here that the histogram is useful for doing true color composite balancing.

**Comment by cholmes** [comment]
Looks good. I think my only question is the logic on including this directly in the stac item vs. in a separate file? Like just describe histogram.json and make it an asset that is linked to.

I'm ok with it in stac item, as we've more towards not trying to make a STAC item truly 'human readable', but this is a further step with a bunch of info. So just curious to hear any thoughts on advantages / disadvantages.

**Comment by m-mohr** [comment]
Yes, I also had the feeling that teh historgram is rather long for direct inclusion, but on the other hand it depends on how it's encoded into a file. So in the end I felt pretty neutral about it as it's really just an array of numbers, not something more complex with objects or so.

**Comment by emmanuelmathot** [comment]
I agree with the eventual overload of the item and the readability. I would be also in favor of putting that information in a separate file. However, there is no way to add links in an asset, isn't it?

**Comment by cholmes** [comment]
> I agree with the eventual overload of the item and the readability. I would be also in favor of putting that information in a separate file. However, there is no way to add links in an asset, isn't it?

This took me a few minutes, writing up a whole response. I see now that the histogram field would be used in an asset. I was thinking it would be a histogram for the Item, but I now see that different assets could have different histograms. 

I mean, we could say that the histogram field is an 'href' that links to the the histogram.json. But that does feel less clean for sure.

**Comment by philvarner** [comment]
> I agree with the eventual overload of the item and the readability. I would be also in favor of putting that information in a separate file. However, there is no way to add links in an asset, isn't it?

I think it would be a good idea to add a `links` field to Asset. I think there other use cases like this, e.g., QA bands. Links can also help resolve the balance between STAC objects having metadata that can be searched over, vs. simply having metadata that will consumed.  If we have a way to reference that metadata through links, then it doesn't have to be included in the object, since it's not going to be searched over.

**Comment by m-mohr** [comment]
@philvarner Could you describe the QA band usecase more? Spontaneously it sounds weird to add links to assets...

**Comment by cholmes** [comment]
Yeah, I'm not sure I understand either. How would links in assets work? An asset could use a full 'link' object? And how does it help with STAC having metadata to just be consumed? 

I agree with the sentiment of thinking about enabling STAC Items to not become overloaded with a bunch of non-search data. But the approach I like the most so far is the ['extended metadata' idea](https://gist.github.com/alkamin/3a261b731c98e45aeee2efe42feb7dd2) Though I guess maybe what this is pointing to is that the non-search stuff may often be at the asset level.

**Comment by emmanuelmathot** [comment]
I also like this extended metadata idea. It allows flexibility in the usage of the metadata included directly in the stac item. This way, I could merge this PR and refer to this `extended metadata` extensions as a best practice once ready.

File: README.md (hunk 2 above)
Comment (emmanuelmathot) [reply README.md:165]: added

File: README.md [hunk 3]
```diff
@@ -145,3 +146,31 @@ In the following value definition example, 185 meters must be substracted from t
...
+
```
[the remark this answers is not in the document]
Comment (emmanuelmathot) [reply README.md:166]: added

**Pull request merged**